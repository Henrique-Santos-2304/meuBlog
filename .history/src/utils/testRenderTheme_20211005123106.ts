import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(<ThemeProvider>{children}</ThemeProvider> );
};
