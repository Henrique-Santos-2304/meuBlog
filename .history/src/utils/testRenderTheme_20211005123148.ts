import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactElement): RenderResult => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider> );
};
