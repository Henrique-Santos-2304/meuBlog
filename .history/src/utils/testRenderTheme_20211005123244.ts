import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider> );
};
