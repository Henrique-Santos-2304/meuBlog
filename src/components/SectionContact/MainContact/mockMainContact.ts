export const mockMainContact = {
  dataContact: {
    title: "contato",
    ariaLabel: "contato",
    ariaLabelTitle: "Titulo da seção",
    Input: {
      name: "name",
      placeholder: "Digite seu Nome",
      type: "text",
      msgError: "Digite um nome",
      textLabel: "Nome",
    },
    inputEmail: {
      name: "email",
      placeholder: "Digite seu email",
      type: "email",
      msgError: "Digite um email",
      textLabel: "Email",
    },
    inputMessage: {
      name: "message",
      placeholder: "Digite sua mensagem",
      type: "text",
      msgError: "Digite uma mensagem",
      textLabel: "Mensagem",
    },
    buttonSend: {
      text: "enviar",
      name: "button",
      type: "submit",
    },
    msgSendEmail: {
      msgSucess: "Enviado com sucesso",
      msgError: "Falha no envio",
      ariaLabel: "enviar email",
      msgFormatEmail: "Formato de email inválido",
    },
  },
  formEmailData: {
    services: "string;",
    form: "string;",
    user: "string;",
  },
};
