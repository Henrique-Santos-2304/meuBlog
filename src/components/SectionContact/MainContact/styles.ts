import styled, { css } from "styled-components";
export const Wrapper = styled.section`
  ${({ theme }) => css`
    max-width: ${theme.grid.containerContent};
    margin: 0 auto;
    padding-top: 0.5rem;

    h1 {
      text-align: center;
    }
    @media (min-width: ${theme.media.medium}) {
      padding: 2rem ${theme.spacings.small};
    }
    @media (min-width: ${theme.media.xmedium}) {
      padding: 2rem ${theme.spacings.large};
    }
  `}
`;
