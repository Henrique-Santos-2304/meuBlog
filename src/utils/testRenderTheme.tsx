import { ThemeProvider } from "styled-components";
import { ChakraProvider } from "@chakra-ui/react";

import { render, RenderResult } from "@testing-library/react";
import theme from "styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <ThemeProvider theme={theme}>
      <ChakraProvider> {children}</ChakraProvider>
    </ThemeProvider>
  );
};
