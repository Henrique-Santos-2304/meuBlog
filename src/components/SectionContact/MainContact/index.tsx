import Headings from "components/globalComponents/Headings";
import { contact, formDatas } from "graphql/typesQueries/types";
import FormEmail from "../FormEmail";
import * as S from "./styles";

type propsContact = {
  dataContact: contact;
  formEmailData: formDatas;
};

const MainContact = ({ dataContact, formEmailData }: propsContact) => (
  <>
    <S.Wrapper id="contacts">
      <Headings
        msg={dataContact.title}
        fontSizeText="2rem"
        marginText="0 0 0 3rem"
        aria-label={dataContact.ariaLabelTitle}
      />
      <FormEmail dataContact={dataContact} formEmailData={formEmailData} />
    </S.Wrapper>
  </>
);

export default MainContact;
