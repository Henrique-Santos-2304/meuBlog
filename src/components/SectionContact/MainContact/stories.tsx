import { Story, Meta } from "@storybook/react";
import MainContact from ".";
import { mockMainContact } from "./mockMainContact";

export default {
  title: "MainContact",
  component: MainContact,
} as Meta;

export const Basic: Story = () => <MainContact {...mockMainContact} />;

export const Default: Story = () => <MainContact {...mockMainContact} />;
