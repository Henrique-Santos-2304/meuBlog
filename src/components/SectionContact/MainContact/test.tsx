import MainContact from ".";
import { screen } from "@testing-library/react";
import { renderTheme } from "utils/testRenderTheme";
import { mockMainContact } from "./mockMainContact";

describe("<MainContact />", () => {
  it("should render ", () => {
    const { container } = renderTheme(<MainContact {...mockMainContact} />);
    expect(container).toMatchSnapshot();
  });
  it("should render ", () => {
    renderTheme(<MainContact {...mockMainContact} />);
    const title = screen.getByRole("heading", {
      name: /contato/i,
    });
    expect(title).toBeInTheDocument();
  });
});
