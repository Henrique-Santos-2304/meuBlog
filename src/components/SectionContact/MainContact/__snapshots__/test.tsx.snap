// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainContact /> should render  1`] = `
.c1 {
  position: relative;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  color: #f8f8f8;
  padding: 0;
  margin: 0 0 0 3rem;
  font-size: 2rem;
  font-family: "Roboto",monospace;
}

.c1::before {
  content: "";
  position: absolute;
  top: 50%;
  left: -1.5rem;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  width: 0.4rem;
  height: 90%;
  border-radius: 2rem;
  background: #b7f600;
}

.c2 {
  position: relative;
  max-width: 74rem;
}

.c3 {
  padding: 0.3rem 0.5rem;
  margin: 0rem auto;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 {
  padding: 0.8rem 1.6rem;
  border-bottom: 0.1rem solid #1e1e1e;
  font-size: 1.4rem;
  background: transparent;
  border-radius: 0.7rem;
  text-align: start;
  color: #f8f8f8;
  -webkit-transition: all 0.3s linear;
  transition: all 0.3s linear;
}

.c6::-webkit-input-placeholder {
  color: rgba(0,0,0,0.45);
}

.c6::-moz-placeholder {
  color: rgba(0,0,0,0.45);
}

.c6:-ms-input-placeholder {
  color: rgba(0,0,0,0.45);
}

.c6::placeholder {
  color: rgba(0,0,0,0.45);
}

.c6:hover,
.c6:focus {
  background: rgba(255,255,255,0.1);
  border: 0.1rem solid #b7f600;
}

.c7 {
  padding: 0.8rem;
  overflow: hidden;
  border: 0;
  font-size: 1.4rem;
  font-family: 'Poppins',sans-serif;
  background: rgba(0,0,0,0.1);
  border-radius: 0.6rem;
  -webkit-transition: all 0.3s linear;
  transition: all 0.3s linear;
  color: #f8f8f8;
}

.c7::-webkit-input-placeholder {
  color: rgba(0,0,0,0.45);
}

.c7::-moz-placeholder {
  color: rgba(0,0,0,0.45);
}

.c7:-ms-input-placeholder {
  color: rgba(0,0,0,0.45);
}

.c7::placeholder {
  color: rgba(0,0,0,0.45);
}

.c7:hover,
.c7:focus {
  border: 0.1rem solid #b7f600;
  outline: none;
}

.c8 {
  margin-bottom: 0.3rem;
  padding: 0.5rem 1.6rem;
  background: transparent;
  color: #b7f600;
  font-weight: 700;
  border-radius: 0.5rem;
  border: 0.1rem solid #b7f600;
  -webkit-transition: all 0.4s ease-in-out;
  transition: all 0.4s ease-in-out;
}

.c8:hover {
  color: #f8f8f8;
  border: 0.1rem solid #f8f8f8;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1rem 0.6rem;
  margin: 0.6rem 0;
}

.c5 {
  font-size: 1.2rem;
  margin: 0.4rem;
  color: #f8f8f8;
}

.c0 {
  max-width: 74rem;
  margin: 0 auto;
  padding-top: 0.5rem;
}

.c0 h1 {
  text-align: center;
}

@media (min-width:620px) {
  .c6 {
    width: 24rem;
  }
}

@media (min-width:768px) {
  .c6 {
    font-size: 1.4rem;
  }
}

@media (min-width:620px) {
  .c7 {
    width: 24rem;
  }
}

@media (min-width:768px) {
  .c7 {
    font-size: 1.4rem;
  }
}

@media (min-width:768px) {
  .c5 {
    font-size: 1.4rem;
  }
}

@media (min-width:620px) {
  .c0 {
    padding: 2rem 2.4rem;
  }
}

@media (min-width:768px) {
  .c0 {
    padding: 2rem 4.0rem;
  }
}

<div>
  <section
    class="c0"
    id="contacts"
  >
    <h1
      class="c1"
    >
      contato
    </h1>
    <div
      class="c2"
    >
      
      <form
        class="c3"
      >
        <div
          class="c4"
        >
          <label
            class="c5"
            for="name"
          >
            Nome
          </label>
          <input
            class="c6"
            id="name"
            name="name"
            placeholder="Digite seu Nome"
            type="text"
          />
        </div>
        <div
          class="c4"
        >
          <label
            class="c5"
            for="email"
          >
            Email
          </label>
          <input
            class="c6"
            id="email"
            name="email"
            placeholder="Digite seu email"
            type="email"
          />
        </div>
        <div
          class="c4"
        >
          <label
            class="c5"
            for="message"
          >
            Mensagem
          </label>
          <textarea
            class="c7"
            cols="25"
            id="message"
            name="message"
            placeholder="Digite sua mensagem"
            rows="5"
          />
        </div>
        <input
          class="c8"
          type="submit"
          value="enviar"
        />
      </form>
    </div>
  </section>
</div>
`;
